drivers/ieee802154/Kconfig

```diff
@@ -66,4 +66,48 @@ config IEEE802154_UPIPE_DRV_NAME
 	default "IEEE802154_UPIPE"
 	depends on IEEE802154_UPIPE
 
+config IEEE802154_UPIPE_PROMISCUOUS
```

Comment:
Change the name here. What you have been implementing is not 15.4 promiscuous mode but "hw filtering".

See include/net/ieee802154_radio.h 

With your modifications, the upipe driver could return IEEE802154_HW_FILTER as hw capabilities.